service/storage_service.cc

```diff
@@ -4720,13 +4721,20 @@ future<> storage_service::do_cluster_cleanup() {
 }
 
 future<sstring> storage_service::wait_for_topology_request_completion(utils::UUID id, bool require_entry) {
+    rtlogger.debug("Start waiting for topology request completion (request id {})", id);
     while (true) {
+        auto c = _topology_state_machine.reload_count;
         auto [done, error] = co_await  _sys_ks.local().get_topology_request_state(id, require_entry);
         if (done) {
             co_return error;
```

Comment:
consider also adding a debug log here to know that we are done ... (although in that case we'd likely see the next logs after returning from here, so can be fine even without)